packages/trade/exchange-irec/test/import.e2e-spec.ts

```diff
@@ -0,0 +1,47 @@
+import request from 'supertest';
+import { expect } from 'chai';
+import { HttpStatus, INestApplication } from '@nestjs/common';
+import { TestUser } from '@energyweb/issuer-irec-api/dist/js/test/issuer-irec-api';
+import { CERTIFICATES_TABLE_NAME } from '@energyweb/issuer-irec-api';
+import { DatabaseService } from '@energyweb/origin-backend-utils';
+
+import { bootstrapTestInstance } from './exchange';
+
+describe('import tests', () => {
+    let app: INestApplication;
+    let databaseService: DatabaseService;
+
+    before(async () => {
+        ({ databaseService, app } = await bootstrapTestInstance());
+
+        await app.init();
+    });
+
+    afterEach(async () => {
+        await databaseService.truncate(CERTIFICATES_TABLE_NAME);
+    });
+
+    after(async () => {
+        await databaseService.truncate(CERTIFICATES_TABLE_NAME);
+        await databaseService.cleanUp();
+        await app.close();
+    });
+
+    it('should import certificate', async () => {
+        const {
+            body: [certificateToImport]
+        } = await request(app.getHttpServer())
+            .get(`/irec/import`)
+            .set({ 'test-user': TestUser.OrganizationDeviceManager })
+            .expect(HttpStatus.OK);
+
+        expect(certificateToImport.asset).to.equal('test-asset-id');
+        expect(certificateToImport.isDeviceImported).to.equal(true);
```

Comment:
What's this `isDeviceImported` flag?

If this is `true`, then we are re-importing an already imported certificate in line 41.

Reply:
This flag says whether Origin knows about the device of this certificate. If the device exists on IREC side only, then we prohibit importing certificate and want user to import the device before importing the certificate

Reply:
Oh, I mixed up devices and certificates.

Yes, it makes total sense now - thanks.